File: MagicSquare/COM+ version/Tutorials/C++ tutorials/Tutorial27.cpp
/* ----------------------------------------------------------------
 * Tutorial 27
 *
 * This tutorial shows how to encrypt and set the password required for opening a document
 * ----------------------------------------------------------------- */

#include "stdafx.h"
#include "EasyXLS.h"
#include <conio.h>


int _tmain(int argc, _TCHAR* argv[])
{
	printf("Tutorial 27\n----------\n");

	HRESULT hr;

	//Initialize COM
	hr = CoInitialize(0);

	// Use the SUCCEEDED macro and get a pointer to the interface
	if(SUCCEEDED(hr))
	{
		//Create a pointer to the interface that generates Excel files
		EasyXLS::IExcelDocumentPtr xls;
		hr = CoCreateInstance(__uuidof(EasyXLS::ExcelDocument),
		NULL,
		CLSCTX_ALL,
		__uuidof(EasyXLS::IExcelDocument),
		(void**) &xls) ;

		if(SUCCEEDED(hr)){
			
			//Create the worksheets
			xls->easy_addWorksheet_2("First tab");
			xls->easy_addWorksheet_2("Second tab");

			//Set the password required for opening the document
			xls->easy_getOptions()->setPasswordToOpen("password");

		
			//Generate the file
			printf("Writing file C:\\Samples\\Tutorial27.xls.");
			xls->easy_WriteXLSFile("C:\\Samples\\Tutorial27.xls");
			
			//Confirm generation
			_bstr_t sError = xls->easy_getError();
			if (strcmp(sError, "") == 0){
				printf("\nFile successfully created. Press Enter to Exit...");
			}
			else{
				printf("\nError encountered: %s", (LPCSTR)sError); 
			}
			
			//Dispose memory
			xls->Dispose();
		}
		else{
			printf("Object is not available!");
		}
	}
	else{
		printf("COM can't be initialized!");
	}

	// Uninitialize COM
	CoUninitialize();

	_getch();
	return 0;
}

File: MagicSquare/COM+ version/Tutorials/PHP tutorials/Tutorial27.php
<?php
	/*==========================================================================
	 | Tutorial 27
	 |
	 | This tutorial shows how to encrypt and set the password required for opening a document
	  ==========================================================================*/

	header("Content-Type: text/html");
	
	echo "Tutorial 27<br>";
	echo "----------<br>";
	


	//Create an instance of the object that generates Excel files
	$xls = new COM("EasyXLS.ExcelDocument");
	

	//Create the worksheets 
	$xls->easy_addWorksheet_2("First tab");
	$xls->easy_addWorksheet_2("Second tab");

	//Set the password required for opening the document
	$xls->easy_getOptions()->setPasswordToOpen("password");
		
	//Generate the file
	echo "Writing file: C:\Samples\Tutorial27.xls<br>";
	$xls->easy_WriteXLSFile("C:\Samples\Tutorial27.xls");
	
	//Confirm generation
	if ($xls->easy_getError() == "")
		echo "File successfully created.";
	else
		echo "Error encountered: " . $xls->easy_getError();
		
	//Dispose memory
	$xls->Dispose();
?>
